google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java

```diff
@@ -0,0 +1,427 @@
+
+    // Creates the campaign budget service client.
+    try (CampaignBudgetServiceClient campaignBudgetServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignBudgetServiceClient()) {
+      // Adds the campaign budget.
+      MutateCampaignBudgetsResponse response =
+          campaignBudgetServiceClient.mutateCampaignBudgets(
+              Long.toString(customerId), ImmutableList.of(operation));
+      // Prints and returns the response.
+      String budgetResourceName = response.getResults(0).getResourceName();
+      System.out.printf("Created campaign budget with resource name '%s'.%n", budgetResourceName);
+      return budgetResourceName;
+    }
+  }
+
+  /**
+   * Creates an App campaign under the given customer ID.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @param budgetResourceName the resource name of the budget to associate with the campaign.
+   * @return the resource name of the newly created App campaign.
+   */
+  private String createCampaign(
+      GoogleAdsClient googleAdsClient, long customerId, String budgetResourceName) {
+    // Creates a campaign.
+    Campaign campaign =
+        Campaign.newBuilder()
+            .setName(StringValue.of("Interplanetary Cruise App #" + System.currentTimeMillis()))
+            .setCampaignBudget(StringValue.of(budgetResourceName))
+            // Recommendation: Set the campaign to PAUSED when creating it to prevent
+            // the ads from immediately serving. Set to ENABLED once you've added
+            // targeting and the ads are ready to serve.
+            .setStatus(CampaignStatus.PAUSED)
+            // All App campaigns have an advertising_channel_type of
+            // MULTI_CHANNEL to reflect the fact that ads from these campaigns are
+            // eligible to appear on multiple channels.
+            .setAdvertisingChannelType(AdvertisingChannelType.MULTI_CHANNEL)
+            .setAdvertisingChannelSubType(AdvertisingChannelSubType.APP_CAMPAIGN)
+            // Sets the target CPA to $1 / app install.
+            //
+            // campaign_bidding_strategy is a 'oneof' message so setting target_cpa
+            // is mutually exclusive with other bidding strategies such as
+            // manual_cpc, commission, maximize_conversions, etc.
+            // See https://developers.google.com/google-ads/api/reference/rpc
+            // under current version / resources / Campaign.
+            .setTargetCpa(TargetCpa.newBuilder().setTargetCpaMicros(Int64Value.of(1000000)).build())
+            // Sets the App campaign settings.
+            .setAppCampaignSetting(
+                AppCampaignSetting.newBuilder()
+                    .setAppId(StringValue.of("com.google.android.apps.adwords"))
+                    .setAppStore(AppCampaignAppStore.GOOGLE_APP_STORE)
+                    .setBiddingStrategyGoalType(
```

Comment:
A comment is missing "Optional: Optimize this campaign for getting new users for your app."